python/mxnet/gluon/rnn/rnn_cell.py

```diff
@@ -784,3 +792,378 @@ def unroll(self, length, inputs, begin_state=None, layout='NTC', merge_outputs=N
+                            name=prefix+'i2h')
+        h2h = F.Convolution(data=states[0],
+                            num_filter=self._hidden_size*self._num_gates,
+                            kernel=self._h2h_kernel,
+                            dilate=self._h2h_dilate,
+                            pad=self._h2h_pad,
+                            stride=tuple([1]*len(self._h2h_kernel)),
+                            weight=h2h_weight,
+                            bias=h2h_bias,
+                            layout=self._conv_layout,
+                            name=prefix+'h2h')
+        return i2h, h2h
+
+    def state_info(self, batch_size=0):
+        raise NotImplementedError("BaseConvRNNCell is abstract class for convolutional RNN")
+
+    def hybrid_forward(self, F, inputs, states):
+        raise NotImplementedError("BaseConvRNNCell is abstract class for convolutional RNN")
+
+class ConvRNNCell(BaseConvRNNCell):
+    """Convolutional RNN cells
+
+    Parameters
+    ----------
+    hidden_size : int
+        Number of units in output symbol.
+    conv_layout : str, default 'NCHW'
+        Layout of ConvolutionOp
+    input_shape : tuple, default (3, 224, 224)
+        Input data tensor shape, with the same layout as conv_layout, excluding batch size.
+    h2h_kernel : tuple of int, default (3, 3)
+        Kernel of Convolution operator in state-to-state transitions.
+    h2h_dilate : tuple of int, default (1, 1)
+        Dilation of Convolution operator in state-to-state transitions.
+    i2h_kernel : tuple of int, default (3, 3)
+        Kernel of Convolution operator in input-to-state transitions.
+    i2h_stride : tuple of int, default (1, 1)
+        Stride of Convolution operator in input-to-state transitions.
+    i2h_pad : tuple of int, default (1, 1)
+        Pad of Convolution operator in input-to-state transitions.
+    i2h_dilate : tuple of int, default (1, 1)
+        Dilation of Convolution operator in input-to-state transitions.
+    i2h_weight_initializer : str or Initializer
+        Initializer for the input weights matrix, used for the convolution
+        transformation of the inputs.
+    h2h_weight_initializer : str or Initializer
+        Initializer for the recurrent weights matrix, used for the convolution
+        transformation of the recurrent state.
+    i2h_bias_initializer : str or Initializer, default zeros
+        Initializer for the bias vector.
+    h2h_bias_initializer : str or Initializer, default zeros
+        Initializer for the bias vector.
+    activation : str or Block
+        default LeakyReLU(alpha=0.2)
+        Type of activation function.
+    prefix : str, default 'conv_rnn_'
+        Prefix for name of layers (and name of weight if params is None).
+    params : RNNParams, default None
+        Container for weight sharing between cells. Created if None.
+    """
+    def __init__(self, hidden_size, conv_layout='NCHW', input_shape=(3, 224, 224),
```

Comment:
input_shape shouldn't be consted like this. layout should be put at the end. put activation in the front